bool uploading
string file_name
string error